internal/testutil/kube_server_compatibility.go

```diff
@@ -44,3 +49,50 @@ func KubeServerMinorVersionInBetweenInclusive(t *testing.T, discoveryClient disc
 
 	return minor >= min && minor <= max
 }
+
+func ConvertMap[T1, T2 comparable, U1, U2 any](m1 map[T1]U1, fT func(T1) T2, fU func(U1) U2) map[T2]U2 {
```

Comment:
Wow, clever. 😄

Should this be a package private function though?

Reply:
Yes, will make that change.